libbeat/docs/shared-autodiscover.asciidoc

```diff
@@ -360,6 +360,103 @@ include::../../x-pack/libbeat/docs/aws-credentials-config.asciidoc[]
 
 endif::autodiscoverAWSELB[]
 
+ifdef::autodiscoverNomad[]
+[float]
+===== Nomad
+
+*Note: This provider is experimental*
+
+The Nomad autodiscover provider watches for Nomad jobs to start, update, and stop.
+
+These are the available fields during within config templating. The `meta.*` fields will be available
+on each emitted event.
+
+* host
+* id
+* meta.name
+* meta.job
+* meta.namespace
+* meta.datacenters
+* meta.region
+* meta.type
+* meta.alloc_id
+* meta.status
+* meta.task.name
+* meta.task.service.name
+* meta.task.service.tags
+* meta.task.service.canary_tags
+
+If the `include_labels` config is added to the provider config, then the list of labels present in
+the config will be added to the event.
+
+If the `exclude_labels` config is added to the provider config, then the list of labels present in
+the config will be excluded from the event.
+
+if the `labels.dedot` config is set to be `true` in the provider config, then `.` in labels will be
+replaced with `_`.
+
+For example:
+
+[source,yaml]
+-------------------------------------------------------------------------------------
+{
+  ...
+  "region": "europe",
+  "type": "service",
+  "alloc_id": "35eba07f-e5e4-20ac-6def-85117bee6efb",
+  "datacenters": [
+    "europe-west4"
+  ],
+  "namespace": "default",
+  "name": "coffeshop.api[0]",
+  "job": "coffeshop"
+  "task": {
+    "service": {
+      "name": [
+        "coffeshop"
+      ],
+      "tags": [
+        "coffeshop",
+        "nginx"
+      ],
+      "canary_tags": [
+        "coffeshop"
+      ]
+    },
+    "name": "api"
+  },
+  ...
+}
```

Comment:
Nomad fields will need to be documented. Could you please add a `fields.yml` file in `x-pack/libbeat/processors/add_nomad_metadata/_meta/` with the fields definitions?
You can see some examples in `x-pack/libbeat/processors/add_cloudfoundry_metadata/_meta/fields.yml` or `libbeat/processors/add_kubernetes_metadata/_meta/fields.yml`.